src/humancellatlas/data/metadata/helpers/staging_area.py

```diff
@@ -0,0 +1,320 @@
+from collections import (
+    defaultdict,
+)
+from dataclasses import (
+    dataclass,
+)
+import json
+import logging
+import os
+from typing import (
+    ClassVar,
+    List,
+    Mapping,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+)
+from uuid import (
+    UUID,
+    uuid5,
+)
+
+from boltons.cacheutils import (
+    cachedproperty,
+)
+from furl import (
+    furl,
+)
+from github import (
+    Github,
+)
+from github.Repository import (
+    Repository,
+)
+
+from humancellatlas.data.metadata.api import (
+    JSON,
+)
+from humancellatlas.data.metadata.helpers.exception import (
+    reject,
+    require,
+)
+from humancellatlas.data.metadata.helpers.schema_validation import (
+    SchemaValidator
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class JsonFile:
+    """
+    A JSON file in the staging area.
+    """
+    uuid: str
+    version: str
+    name: str
+    content: JSON
+
+    def __post_init__(self):
+        pass  # self.validator.validate_json(self.content, self.name)
+
+    @cachedproperty
+    def validator(self):
+        return SchemaValidator()
```

Comment:
The validator can be a singleton and we definitely don't want an instance of it per instance of this class or its sub classes.

Take my word for it (and I've said this before) `@classmethod` for anything but factories is a bad smell.